x-pack/plugins/cases/server/client/user_actions/users.ts

```diff
@@ -54,20 +54,22 @@ export const getUsers = async (
       .filter((participant) => participant.user.profile_uid != null)
       .map((participant) => participant.user.profile_uid) as string[];
 
-    const userProfileUids = new Set([...assignees, ...participantsUids]);
+    /**
+     * To avoid duplicates, a user that is already participant
+     * should be removed from the assignees returned by the case.
+     */
+
+    const assigneesUids = theCase.case.assignees
+      .map((assignee) => assignee.uid)
+      .filter((uid) => !participantsUids.includes(uid));
+
+    const userProfileUids = new Set([...users, ...participantsUids, ...assigneesUids]);
     const userProfiles = await getUserProfiles(securityStartPlugin, userProfileUids);
 
     for (const participant of participants) {
-      const user =
-        participant.user.profile_uid != null
-          ? getUserInformation(userProfiles, participant.user.profile_uid)
-          : {
-              email: participant.user.email,
-              full_name: participant.user.full_name,
-              username: participant.user.username,
-            };
-
-      users.push({
+      const user = getUserInformation(userProfiles, participant.user.profile_uid, participant.user);
```

Comment:
super nit: if you wanted to reduce the loops we could extract one to a function that looks something like pseudo code:

```
const convertUserInfoToResponse = (userProfiles, Array<{ profileUid, user }>, type) => {
  const response = [];
  for (const info of usersInfoArray) {
    response.push({ user: getUserInformation(userProfiles, info.profileUid, info.user), type });
  }
  
  return response;
}
```

Then these loops would look like:

```
const usersResponse = [
  ...convertUserInfoToResponse(participants),
  ...convertUserInfoToResponse(assigneesUids),
  ...convertUserInfoToResponse(users.values()),
]
```